inbm-v5.md

```diff
@@ -0,0 +1,335 @@
+
+- the primary OTA feature set that INBM provided remain the same i.e.:
+    - OS Update
+    - Firmware Update
+    - Application update
+    - Basic telemetry and events reporting
+
+- the primary `device-management` interfaces used and provided by INBM remain the same i.e.:
+    - `inbc`, command-line interface for local usage
+    - Azure IOT Central connectivity for `CSP` enablement
+    - ThingsBoard connectivity for `on-premise` device management 
+ 
+> **NOTE** The availability of these features shall be staged in multiple releases starting with INBM v5.0
+
+## Architecture Diagram
+
+Below is a high-level architecture diagram for INBMv5 leveraging Golang's `multi-threading` capability and `channel` based inter-thread communication.
+
+![INBMv5 High-Level Architecture](./INBMv5-architecture.svg)
+
+   Figure 1: INBMv5 High-Level Architecture
+
+### Key Components
+
+1. #### inbm-daemon
+   - **Function**: Main manageability application which runs in the background
+   - **Main Tasks**: 
+      - Spawns other `persistant` or `long-living` threads like `cloud-client`, `dispatcher-queue`, `telemetry-reporter` and `aggregator`. 
+      - Acts as a server and accepts incoming requests from `inbc` and `cloud-connect` over unix socket and pushes the over-the-air update commands to dispatcher-queue. 
+
+1. #### inbc
+   - **Function**: In-band manageability's commandline interface
+   - **Main Tasks**: 
+      - `inbc` acts as the commandline interace to other `previlaged` user-space applications to perform device-management actions (like os updates or firmware update etc) on the underlying host.
+      - a `trusted client` application which communicates with `inbm-daemon` over unix-sockets, translating manageability commands into gRPC API calls.
+   - **Example Use**: 
+        ``` 
+        inbc sota {--uri, -u=URI} 
+        [--releasedate, -r RELEASE_DATE; default="2026-12-31"] 
+        [--username, -un USERNAME]
+        [--mode, -m MODE; default="full", choices=["full","no-download", "download-only"] ]
+        [--reboot, -rb; default=yes]
+        [--package-list, -p=PACKAGES]
+        ```
+    For detailed usage of `inbc` refer to ![inbc usage guide](inbc-readme-link)
+
+1. #### cloud-client
+   - **Function**: Cloud `device management service` (DMS) connecting thread
+   - **Main Tasks**:
+      - North-bound acts MQTT client connecting to DMS (e.g. Azure IOT Central or ThingsBoard)
+      - South-bound acts as `inbm-daemon` client translating over-the-air (ota) commands from DMS to `inbm-daemon` gRPC API's
+
+1. #### dispatcher-queue
+   - **Function**: Management command queue
+   - **Main Tasks**: 
+      - implements a simple queue of size `1` for device management commands
+      - invokes `updater` thread based on the type of update command e.g. firmware or os or application
```

Comment:
also handles post-reboot tasks by launching 'updater' thread a second time if we have the appropriate entry in our metadata.  also: we could keep a metadata file like platform update agent does, similar to our existing `dispatcher_state`

Reply:
Added this to inbm-daemon's function.